src/main/java/com/buyanywhere/productcatalog/models/Category.java

```diff
@@ -0,0 +1,42 @@
+package com.buyanywhere.productcatalog.models;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+
+@Entity
+public class Category{
+
+    @Id
+    @GeneratedValue
+    private long id;
```

Comment:
In the repository you are using Long. Please replace all occurrences of long with Long